src/main/java/org/zanata/mt/api/TranslateResource.java

```diff
@@ -0,0 +1,274 @@
+package org.zanata.mt.api;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.zanata.mt.api.dto.APIErrorResponse;
+import org.zanata.mt.api.dto.Article;
+import org.zanata.mt.api.dto.LocaleId;
+import org.zanata.mt.dao.DocumentDAO;
+import org.zanata.mt.dao.LocaleDAO;
+import org.zanata.mt.exception.BadTranslationRequestException;
+import org.zanata.mt.exception.TranslationEngineException;
+import org.zanata.mt.model.Locale;
+import org.zanata.mt.model.Provider;
+import org.zanata.mt.util.TranslationUtil;
+import org.zanata.mt.service.TranslationService;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import org.zanata.mt.util.UrlUtil;
+
+@Path("/translate")
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+@RequestScoped
+public class TranslateResource {
+    private static final Logger log =
+        LoggerFactory.getLogger(TranslateResource.class);
+
+    @Inject
+    private TranslationService translationService;
+
+    @Inject
+    private LocaleDAO localeDAO;
+
+    @Inject
+    private DocumentDAO documentDAO;
+
+    /**
+     * Default Machine translation provider: {@link Provider#MS}
+     */
+    private static final String DEFAULT_PROVIDER = "MS";
+
+    /**
+     * Locale mapping
+     *
+     * TODO: make this configurable, at the moment it is done manually.
+     */
+    private static final HashMap<LocaleId, LocaleId> LocaleMap = Maps.newHashMap();
+    static
+    {
+        LocaleMap.put(new LocaleId("zh-hans"), LocaleId.ZH_CN);
+    }
+    private LocaleId getLocaleFromMap(LocaleId locale) {
+        return LocaleMap.containsKey(locale) ? LocaleMap.get(locale) : locale;
+    }
+
+    @SuppressWarnings("unused")
+    public TranslateResource() {
+    }
+
+    @VisibleForTesting
+    TranslateResource(TranslationService translationService,
```

Comment:
Since you are using  CDI now, you could simply add constructor injection and it wouldn't need the `@VisibleForTesting`